### PR TITLE
ENH: Add package testing `GitHub Actions` workflow

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add package testing `GitHub Actions` workflow.

Add the corresponding badge to the `README` file.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@thewtex @grlee77 @tbirdso Let me know if there is a standardized testing skeleton that is being used in pure Python ITK packages so that I can adapt this.

File: .github/workflows/build-test-package-python.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      # max-parallel: 6
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (thewtex) [inline .github/workflows/build-test-package-python.yml]: Can we add 3.11?

**Comment by thewtex** [comment]
@jhlegarreta thanks!

We need to enable a GPU-based hosted runner. I will work on this later this week.

xref #1

File: .github/workflows/build-test-package-python.yml (hunk 1 above)
Comment (jhlegarreta) [reply .github/workflows/build-test-package-python.yml]: Done in [2f2e814](https://github.com/InsightSoftwareConsortium/itk_cucim/commit/2f2e814bd3914bfebbeddcba2eb64d9ab16c461d).

**Comment by jhlegarreta** [comment]
> We need to enable a GPU-based hosted runner. I will work on this later this week.

Yep, I later realized that this would require a GPU setup/would not work as-is on the instances available from GitHub. Sorry Matt.

**Comment by thewtex** [comment]
We are not getting test builds with the forked pull request, so I am going to re-open with a branch on the repository.

**Comment by thewtex** [comment]
Testing in https://github.com/InsightSoftwareConsortium/itk_cucim/pull/23

**Pull request closed without merging**